addons/mass_mailing/models/mailing.py

```diff
@@ -233,7 +232,9 @@ def _compute_mailing_domain(self):
     @api.model
     def create(self, values):
         if values.get('subject') and not values.get('name'):
-            values['name'] = "%s %s" % (values['subject'], datetime.strftime(fields.datetime.now(), tools.DEFAULT_SERVER_DATETIME_FORMAT))
+            timezone = self._context.get('tz') or self.env.user.partner_id.tz or 'UTC'
+            dt_value = fields.Datetime.context_timestamp(self.with_context(tz=timezone), fields.datetime.now())
+            values['name'] = "%s %s %s" % (values['subject'], format_date(self.env, dt_value), format_time(self.env, dt_value, time_format=False))
```

Comment:
Just got to know we have `format_datetime` from v13
https://github.com/odoo/odoo/blob/aa98750fa4f9b1278eaf7d656c0c1a8c81f86325/odoo/tools/misc.py#L1265